easypy/decorations.py

```diff
@@ -73,34 +73,34 @@ def inner(func=None, **kwargs):
 def singleton_contextmanager(func):
     from .caching import locking_cache
 
+    class CtxManager():
```

Comment:
I moved it outside the `inner` class - not sure why was it even there in the first place.

Should it be moved to the global scope entirely?

Reply:
perhaps... "ReentrantContextManager"?...
Does it have any justification on its own?

Reply:
The name of the class should follow the name of the function - so `SingletonContextManager`, and if we change the name later we'll change the name of the class as well.

As for justifications - I don't think one needs justifications to have a class in the global scope. This should be the default approach, and you need a special reason to declare a class inside a function.